pkg/kubelet/kubelet.go

```diff
@@ -2010,6 +2010,10 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 
 	// Wait for volumes to attach/mount
 	if err := kl.volumeManager.WaitForAttachAndMount(ctx, pod); err != nil {
+		if errors.Is(err, volumemanager.ErrVolumeAttachLimitExceeded) {
```

Comment:
Can we also record the reason via `recordAdmissionRejection` ?

Reply:
hmm this doesn't immediately seem like an admission rejection to me, this seems pretty far into the creation process to be considered admission...

Reply:
I wouldn't call myself an expert here so I'd defer to others opinions but I would personally not record it as an admission rejection

Reply:
hmm, that may be true. Is it safe to reject pods here then? The reason I asked here to record the metric is, because everytime we call `rejectPod`, we make sure to call `recordAdmissionRejection`. All `recordAdmissionRejection` does is - it records a metric.

There are already other resource exhausted errors which can cause rejection in kubelet. To me, volume limit exhaustion doesn't seem all that different.  We haven't started any of associated containers yet and hence it should be safe to fail the pod?

Reply:
Sergey recommended updating recordAdmissionRejection to include the extra metric label value, resolving.